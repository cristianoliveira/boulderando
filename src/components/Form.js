import { useState } from 'react'
import styled from '@emotion/styled'

import FormControl from '@mui/material/FormControl'
import FormHelperText from '@mui/material/FormHelperText'
import Button from '@mui/material/Button'
import Grid from '@mui/material/Grid'
import { useForm } from 'react-hook-form'
import Alert from '@mui/material/Alert'
import AlertTitle from '@mui/material/AlertTitle'

import TextField from './TextField'
import DateField from './DateField'

const StyledForm = styled.form`
  padding: 20px;
`

const FULL_GRID = 12

const showErrorFields = errors => Object.keys(errors).map(e =>e).join(', ')

function Form({ onSubmit }) {
  const {
    register,
    handleSubmit,
    formState: { errors, isValid, isSubmitted },
  } = useForm()

  return (
    <StyledForm onSubmit={handleSubmit(onSubmit)}>
      <Grid
        direction="row"
        justify="flex-start"
        alignItems="flex-start"
        spacing={2}
        container
      >
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            name="name"
            label="First Name"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID / 2}>
          <TextField name="last_name" label="Last Name" register={register} />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <DateField name="birthday" label="Birthday" register={register} />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <TextField
            name="address"
            label="Street address"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            name="postal_code"
            label="Postal Code"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            name="city"
            label="City"
            value="Berlin"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <TextField name="email" label="Email" register={register} />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <TextField
            name="usc_number"
            label="Urban Sports Club No."
            register={register}
          />
          <FormHelperText>Your USB member number.</FormHelperText>
        </Grid>
        <Grid item xs={FULL_GRID}>
          {isSubmitted && !isValid && (
            <Alert severity="error">All fields are required. Missing fields: {showErrorFields(errors)}</Alert>
          )}
        </Grid>
      </Grid>
      <FormControl fullWidth>
        <Button variant="contained" type="submit">
          Save data
        </Button>
      </FormControl>
      <FormHelperText variant="outlined">
        ** This data is stored in your browser and can be deleted at any time
      </FormHelperText>
    </StyledForm>
  )
}

export default Form
