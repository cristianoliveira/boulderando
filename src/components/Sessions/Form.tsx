import styled from '@emotion/styled'

import FormControl from '@mui/material/FormControl'
import FormHelperText from '@mui/material/FormHelperText'
import Button from '@mui/material/Button'
import Grid from '@mui/material/Grid'
import { useForm } from 'react-hook-form'
import Alert from '@mui/material/Alert'

import SelectField from '../fields/SelectField'

import * as TID from '../../constants/data-testid'

const StyledForm = styled.form`
  padding: 20px;
`

const FULL_GRID = 12

const showErrorFields = (errors: object): string =>
  Object.keys(errors)
    .map((e) => e)
    .join(', ')

// Sometime in life you have to just have faith
// range from 8:00 to 21:30
const timesSelection: Array<string> = [...Array(20).keys()]
  .filter((i) => i >= 8 && i <= 20)
  .map((i) => [`${i}:00 - ${i + 2}:00`, `${i}:30 - ${i + 2}:30`])
  .flat()

function Form({ onSubmit }: Submitable): JSX.Element {
  const {
    register,
    handleSubmit,
    formState: { errors, isValid, isSubmitted },
  } = useForm({})

  const errorsList = showErrorFields(errors)
  return (
    <StyledForm
      onSubmit={handleSubmit((data) =>
        onSubmit({ ...data, day_of_week: data.human_date.replace(/this /, '') })
      )}
    >
      <Grid
        direction="row"
        alignItems="flex-start"
        spacing={2}
        container
      >
        <Grid item xs={FULL_GRID}>
          <SelectField
            data-testid={TID.CUSTOM_SESSION_FORM_GYM}
            name="gym_name"
            label="Gym"
            register={register}
            selectOptions={[
              { label: 'Boulderklub', value: 'boulderklub' },
              { label: 'Basement', value: 'basement' },
            ]}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <SelectField
            data-testid={TID.CUSTOM_SESSION_FORM_DAY}
            name="human_date"
            label="Day of the week"
            register={register}
            selectOptions={[
              { label: 'Monday', value: 'this monday' },
              { label: 'Tuesday', value: 'this tuesday' },
              { label: 'Wednesday', value: 'this wednesday' },
              { label: 'Thursday', value: 'this thursday' },
              { label: 'Friday', value: 'this friday' },
              { label: 'Saturday', value: 'this saturday' },
            ]}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <SelectField
            data-testid={TID.CUSTOM_SESSION_FORM_TIME}
            name="time"
            label="Time"
            register={register}
            selectOptions={timesSelection.map((time) => ({
              label: time,
              value: time,
            }))}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          {isSubmitted && !isValid && errorsList && (
            <Alert
              data-testid={TID.USER_FORM_ERROR_MESSAGE_CONTAINER}
              severity="error"
            >
              All fields are required. Missing fields: {errorsList}
            </Alert>
          )}
        </Grid>
      </Grid>
      <FormControl fullWidth>
        <Button
          data-testid={TID.CUSTOM_SESSION_FORM_SUBMIT_BUTTON}
          variant="contained"
          type="submit"
        >
          Save data
        </Button>
      </FormControl>
      <FormHelperText variant="outlined">
        ** This data is stored in your browser and can be deleted at any time
      </FormHelperText>
    </StyledForm>
  )
}

export default Form
