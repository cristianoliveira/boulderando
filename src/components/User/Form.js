import { useState } from 'react'
import styled from '@emotion/styled'

import FormControl from '@mui/material/FormControl'
import FormHelperText from '@mui/material/FormHelperText'
import Button from '@mui/material/Button'
import Grid from '@mui/material/Grid'
import { useForm } from 'react-hook-form'
import Alert from '@mui/material/Alert'
import AlertTitle from '@mui/material/AlertTitle'

import PhoneField from '../fields/PhoneField'
import TextField from '../fields/TextField'
import DateField from '../fields/DateField'

import * as TID from '../../constants/data-testid'

const StyledForm = styled.form`
  padding: 20px;
`

const FULL_GRID = 12

const showErrorFields = (errors) =>
  Object.keys(errors)
    .map((e) => e)
    .join(', ')

function Form({ user = {}, onSubmit }) {
  const {
    register,
    handleSubmit,
    formState: { errors, isValid, isSubmitted },
  } = useForm({
    defaultValues: { type: 'Urban Sports Club', ...user },
  })

  const errorsList = showErrorFields(errors)
  return (
    <StyledForm onSubmit={handleSubmit(onSubmit)}>
      <Grid
        direction="row"
        justify="flex-start"
        alignItems="flex-start"
        spacing={2}
        container
      >
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            data-testid={TID.USER_INPUT_FIRST_NAME}
            name="name"
            label="First Name"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            data-testid={TID.USER_INPUT_LAST_NAME}
            name="last_name"
            label="Last Name"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <DateField
            data-testid={TID.USER_INPUT_BIRTHDAY}
            name="birthday"
            label="Birthday"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <TextField
            data-testid={TID.USER_INPUT_STREET_ADDRESS}
            name="address"
            label="Street address"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            data-testid={TID.USER_INPUT_POSTALCODE}
            name="postal_code"
            label="Postal Code"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            data-testid={TID.USER_INPUT_CITY}
            name="city"
            label="City"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <TextField
            data-testid={TID.USER_INPUT_EMAIL}
            name="email"
            label="Email"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <PhoneField
            data-testid={TID.USER_INPUT_PHONE_NUMBER}
            name="phone_number"
            label="Phone Number"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <div style={{ visibility: 'hidden', height: 0 }}>
            <TextField name="type" register={register} />
          </div>
          <TextField
            data-testid={TID.USER_INPUT_URBAN_SPORT_NUMBER}
            name="usc_number"
            label="Urban Sports Club No."
            register={register}
          />
          <FormHelperText>Your USB member number.</FormHelperText>
        </Grid>
        <Grid item xs={FULL_GRID}>
          {isSubmitted && !isValid && errorsList && (
            <Alert
            data-testid={TID.USER_FORM_ERROR_MESSAGE_CONTAINER}
            severity="error">
              All fields are required. Missing fields: {errorsList}
            </Alert>
          )}
        </Grid>
      </Grid>
      <FormControl fullWidth>
        <Button
          data-testid={TID.USER_FORM_SUBMIT_BUTTON}
          variant="contained"
          type="submit"
        >
          Save data
        </Button>
      </FormControl>
      <FormHelperText variant="outlined">
        ** This data is stored in your browser and can be deleted at any time
      </FormHelperText>
    </StyledForm>
  )
}

export default Form
