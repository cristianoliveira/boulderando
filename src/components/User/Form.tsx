import styled from '@emotion/styled'

import { useForm, FieldError } from 'react-hook-form'

import FormControl from '@mui/material/FormControl'
import FormHelperText from '@mui/material/FormHelperText'
import Button from '@mui/material/Button'
import Grid from '@mui/material/Grid'
import Alert from '@mui/material/Alert'

import PhoneField from '../fields/PhoneField'
import TextField from '../fields/TextField'
import DateField from '../fields/DateField'

import * as TID from '../../constants/data-testid'

const StyledForm = styled.form`
  padding: 20px;
`

const FULL_GRID = 12

type UserFormProps = {
  user?: User
  telegramIdParam: string
} & Submitable

const showErrorFields = (errors: Partial<FieldError>) =>
  Object.keys(errors)
    .map((e) => e)
    .join(', ')

function Form({ user, telegramIdParam, onSubmit }: UserFormProps) {
  const {
    register,
    handleSubmit,
    setValue,
    formState: { errors, isValid, isSubmitted },
  } = useForm({
    defaultValues: {
      ...user,
      type: 'Urban Sports Club',
    },
  })

  setValue('telegram_id', telegramIdParam || user?.telegram_id)

  const errorsList = showErrorFields(errors as Partial<FieldError>)
  return (
    <StyledForm
      onSubmit={handleSubmit((data) => {
        onSubmit(data)
      })}
    >
      <Grid
        direction="row"
        alignItems="flex-start"
        spacing={2}
        container
      >
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            data-testid={TID.USER_INPUT_FIRST_NAME}
            name="name"
            label="First Name"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            data-testid={TID.USER_INPUT_LAST_NAME}
            name="last_name"
            label="Last Name"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <DateField
            data-testid={TID.USER_INPUT_BIRTHDAY}
            name="birthday"
            label="Birthday"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <TextField
            data-testid={TID.USER_INPUT_STREET_ADDRESS}
            name="address"
            label="Street address"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            data-testid={TID.USER_INPUT_POSTALCODE}
            name="postal_code"
            label="Postal Code"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID / 2}>
          <TextField
            data-testid={TID.USER_INPUT_CITY}
            name="city"
            label="City"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <TextField
            data-testid={TID.USER_INPUT_EMAIL}
            name="email"
            label="Email"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <PhoneField
            data-testid={TID.USER_INPUT_PHONE_NUMBER}
            name="phone_number"
            label="Phone Number"
            register={register}
          />
        </Grid>
        <Grid item xs={FULL_GRID}>
          <div style={{ visibility: 'hidden', height: 0 }}>
            <TextField data-testid="type" name="type" register={register} />
          </div>
          <TextField
            data-testid={TID.USER_INPUT_URBAN_SPORT_NUMBER}
            name="usc_number"
            label="Urban Sports Club No."
            register={register}
          />
          <FormHelperText>Your USB member number.</FormHelperText>
        </Grid>
        <Grid item xs={FULL_GRID}>
          {isSubmitted && !isValid && errorsList && (
            <Alert
              data-testid={TID.USER_FORM_ERROR_MESSAGE_CONTAINER}
              severity="error"
            >
              All fields are required. Missing fields: {errorsList}
            </Alert>
          )}
        </Grid>
      </Grid>
      <FormControl fullWidth>
        <Button
          data-testid={TID.USER_FORM_SUBMIT_BUTTON}
          variant="contained"
          type="submit"
        >
          Save data
        </Button>
      </FormControl>
      <FormHelperText variant="outlined">
        Your personal data is stored only in your browser and can be deleted at
        any time. When scheduling specific sessions, your data is transmitted to
        the corresponding Gym using the Dr. Plano booking system. By scheduling
        sessions you&apos;re automatically agreeing to the Terms and Conditions
        and Privacy Policy of that Gym.
      </FormHelperText>
    </StyledForm>
  )
}

export default Form
