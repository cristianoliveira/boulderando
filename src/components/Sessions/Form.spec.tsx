import React from 'react'
// eslint-disable-next-line
import renderer from 'react-test-renderer'

import Form from './Form'

describe('Form', () => {
  it('matches the snapshot', async () => {
    expect(
      renderer
        .create(
          <Form onSubmit={jest.fn()} />
        )
        .toJSON()
    ).toMatchSnapshot()
  })
})
