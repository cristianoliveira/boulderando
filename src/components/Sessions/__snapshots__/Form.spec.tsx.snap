// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form matches the snapshot 1`] = `
<form
  className="css-gz8dae"
  onSubmit={[Function]}
>
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-lndw90-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth css-zevgu-MuiFormControl-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="gym_name"
        >
          Gym
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          data-testid="custom_session_form_gym"
          onClick={[Function]}
        >
          <div
            aria-describedby="Input for gym_name"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="gym_name"
            className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="gym_name"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            disabled={false}
            name="gym_name"
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-hdw1oc"
            >
              <span
                className="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth css-zevgu-MuiFormControl-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="human_date"
        >
          Day of the week
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          data-testid="custom_session_form_day"
          onClick={[Function]}
        >
          <div
            aria-describedby="Input for human_date"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="human_date"
            className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="human_date"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            disabled={false}
            name="human_date"
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-hdw1oc"
            >
              <span
                className="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    >
      <div
        className="MuiFormControl-root MuiFormControl-marginDense MuiFormControl-fullWidth css-zevgu-MuiFormControl-root"
      >
        <label
          className="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
          data-shrink={false}
          htmlFor="time"
        >
          Time
        </label>
        <div
          className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
          data-testid="custom_session_form_time"
          onClick={[Function]}
        >
          <div
            aria-describedby="Input for time"
            aria-expanded="false"
            aria-haspopup="listbox"
            aria-labelledby="time"
            className="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
            id="time"
            onBlur={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onMouseDown={[Function]}
            role="button"
            tabIndex={0}
          >
            <span
              className="notranslate"
            >
              ​
            </span>
          </div>
          <input
            aria-hidden={true}
            className="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
            disabled={false}
            name="time"
            onAnimationStart={[Function]}
            onChange={[Function]}
            required={false}
            tabIndex={-1}
          />
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
            data-testid="ArrowDropDownIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
          <fieldset
            aria-hidden={true}
            className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="css-hdw1oc"
            >
              <span
                className="notranslate"
              >
                ​
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 css-1idn90j-MuiGrid-root"
    />
  </div>
  <div
    className="MuiFormControl-root MuiFormControl-fullWidth css-q8hpuo-MuiFormControl-root"
  >
    <button
      className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
      data-testid="custom_session_form_submit_button"
      disabled={false}
      onBlur={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="submit"
    >
      Save data
    </button>
  </div>
  <p
    className="MuiFormHelperText-root MuiFormHelperText-contained css-1wc848c-MuiFormHelperText-root"
  >
    ** This data is stored in your browser and can be deleted at any time
  </p>
</form>
`;
